clang/lib/Sema/TreeTransform.h

```diff
@@ -7246,6 +7250,24 @@ QualType TreeTransform<Derived>::TransformDependentBitIntType(
   return Result;
 }
 
+template <typename Derived>
+QualType TreeTransform<Derived>::TransformPredefinedSugarType(
+    TypeLocBuilder &TLB, PredefinedSugarTypeLoc TL) {
+  const PredefinedSugarType *EIT = TL.getTypePtr();
+  QualType Result = TL.getType();
+
+  if (getDerived().AlwaysRebuild()) {
+    Result = getDerived().RebuildPredefinedSugarType(
+        EIT->getKind(), EIT->desugar(), TL.getNameLoc());
+    if (Result.isNull())
+      return QualType();
+  }
```

Comment:
If this had an underlying type, then this would need to transform that as well. But if not, I don't think you should bother transforming anything, just rebuild inline.